File: AtCoder/ABC/195/python/C-Comma.py
N = int(input())
S = str(N)
a = len(S) // 3
b = len(S) % 3
res = 0
if b == 0:
    ub = 1000
    lb = 1
    for i in range(a):
        if i == a - 1:
            res += (N - lb + 1) * i
        else:
            res += i * (ub - lb)
            lb = ub
            ub *= 10**3
else:
    ub = 1000
    lb = 1
    for i in range(a):
        res += i * (ub - lb)
        lb = ub
        ub *= 10**3
        i += 1
    res += (N - lb + 1) * a
print(res)

File: AtCoder/ABC/195/cpp/C-Comma.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll N;

int main() {
  cin >> N;
  string NS = to_string(N);
  ll a = NS.size() / 3;
  ll b = NS.size() % 3;
  ll res = 0;
  ll ub = 1000;
  ll lb = 1;
  if (b == 0) {
    for (ll i = 0; i < a; ++i) {
      if (i == a - 1)
        res += (N - lb + 1) * i;
      else {
        res += i * (ub - lb);
        lb = ub;
        ub *= 1000;
      }
    }
  } else {
    for (ll i = 0; i < a; ++i) {
      res += i * (ub - lb);
      lb = ub;
      ub *= 1000;
    }
    res += (N - lb + 1) * a;
  }
  printf("%lld\n", res);
}